<font size = "5"> Interpreting Logistic Regression Coefficients</font> <br>

In [1]:
library(dplyr)


Attaching package: 'dplyr'


The following objects are masked from 'package:stats':

    filter, lag


The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union




In [2]:
df = readr::read_csv("stroke_data.csv", show_col_types = FALSE)

head(df)                                    

id    gender age hypertension heart_disease ever_married work_type    
1  9046 Male   67  0            1             Yes          Private      
2 31112 Male   80  0            1             Yes          Private      
3 60182 Female 49  0            0             Yes          Private      
4  1665 Female 79  1            0             Yes          Self-employed
5 56669 Male   81  0            0             Yes          Private      
6 53882 Male   74  1            1             Yes          Private      
  Residence_type avg_glucose_level bmi  smoking_status  stroke
1 Urban          228.69            36.6 formerly smoked 1     
2 Rural          105.92            32.5 never smoked    1     
3 Urban          171.23            34.4 smokes          1     
4 Rural          174.12            24.0 never smoked    1     
5 Urban          186.21            29.0 formerly smoked 1     
6 Rural           70.09            27.4 never smoked    1

Train a logistic regression model with a continuous independent variable, a binary independent variable, and an interaction term.

In [3]:
#Running the glm model on age(a continous variable) and heart_disease(a binary variable) and their interaction
log.model = glm(formula = stroke ~ heart_disease*age,
                data = df,
                family = binomial()
                )
summary(log.model)


Call:
glm(formula = stroke ~ heart_disease * age, family = binomial(), 
    data = df)

Coefficients:
                   Estimate Std. Error z value Pr(>|z|)    
(Intercept)       -7.450753   0.385977 -19.304   <2e-16 ***
heart_disease      3.092103   1.357222   2.278   0.0227 *  
age                0.074957   0.005737  13.065   <2e-16 ***
heart_disease:age -0.035850   0.018947  -1.892   0.0585 .  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for binomial family taken to be 1)

    Null deviance: 1728.4  on 4908  degrees of freedom
Residual deviance: 1398.3  on 4905  degrees of freedom
AIC: 1406.3

Number of Fisher Scoring iterations: 7


Interpret each coefficient in relation to the log-odds. 

For the interaction term, plug in zero and one for the binary predictor and interpret the slope of the continuous predictor.

In [4]:
round(coef(log.model), 3)

(Intercept)     heart_disease               age heart_disease:age 
           -7.451             3.092             0.075            -0.036

**b-0:** The intercept coefficient between the heart_disease(binary) and age(continous) represents the log odds of the stroke(the dependent variable) when heart_disease and age are 0  which in our case is -7.45.

**b_1:** When age(continous) is zero, the log odds of having a stroke when heart_disease = 1 is b_1 larger than the log odds of having a stroke when heart_disease = 0.

**b_2:** When heart_disease is 0 (base line), the b_2 coefficient represents the slope of the log odds of having a stroke(the dependent variable). <br>
So, if heart_disease is 0, each one unit change in the independent variable age would increase the log odds of having a stroke by 0.075.

**b_3:** b_3 is the difference between the slopes of the log odds of having a stroke when heart_disease = 1 and when heart_disease = 0. <br>
So, if heart_disease is 1, each one unit change in the independent variable age would increase the log odds of having a stroke 0.036 less than when heart_disease is 0..